bindings/python/YarpUtilities/src/VectorsCollection.cpp

```diff
@@ -65,17 +66,23 @@ void CreateVectorsCollectionClient(pybind11::module& module)
         .def("connect", &VectorsCollectionClient::connect)
         .def("disconnect", &VectorsCollectionClient::disconnect)
         .def("get_metadata",
-        [](VectorsCollectionClient& impl) -> BipedalLocomotion::YarpUtilities::VectorsCollectionMetadata
-        {
-            BipedalLocomotion::YarpUtilities::VectorsCollectionMetadata metadata;
-            impl.getMetadata(metadata);
-            return metadata;
-        })
+             [](VectorsCollectionClient& impl)
+                 -> BipedalLocomotion::YarpUtilities::VectorsCollectionMetadata {
+                 BipedalLocomotion::YarpUtilities::VectorsCollectionMetadata metadata;
+                 impl.getMetadata(metadata);
+                 return metadata;
+             })
         .def("read_data",
-             [](VectorsCollectionClient& impl, bool shouldWait) -> std::map<std::string, std::vector<double>>
-             {
-                BipedalLocomotion::YarpUtilities::VectorsCollection* collection = impl.readData(shouldWait);
-                return collection->vectors;
+             [](VectorsCollectionClient& impl,
+                bool shouldWait) -> std::optional<std::map<std::string, std::vector<double>>> {
+                 BipedalLocomotion::YarpUtilities::VectorsCollection* collection
+                     = impl.readData(shouldWait);
+                 if (collection == nullptr)
+                 {
+                     // Return an empty optional if the collection is not available
+                     return std::nullopt;
+                 }
+                 return collection->vectors;
              });
```

Comment:
You can directly return the connection

Reply:
Done in 4f975ab8fdeff6aec2c5020d51a179a33c600836
Returning the collection made it unnecessary to check if the collection was a `nullptr`. I removed the unnecessary lines. I tested it and did not experience any segmentation faults.